PasswordDatabase#save implementation and unit tests

Majkle opened the pull request:
Michael Pinkas

rodi0878 commented:
- dependency is currently used only in tests and should be scoped to tests only 
- test files must not be in commits/repo - remove them
- also please fix merge conflicts

Majkle commented:
Added test scope for dependency, rebased the commit.
Files are not output of unit tests. Files used for validation of the ouput files. Can they be kept or should the unit tests validate the content of output files in different way?
